### PR TITLE
Converted backend to typescript.

### PR DESCRIPTION
**Pull request by KyleSaunders1310**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -10,4 +10,4 @@ RUN npm install --production
 
 COPY . .
 
-CMD ["node", "app.js"]
```
Comment (Solenoden) [inline Dockerfile:13]: This should target the compiled app.js file. Node can only run Javascript so the Typescript files get compiled into Javascript for Node to use.

**Review by Solenoden — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (KyleSaunders1310) [reply Dockerfile:13]: Fixed it.

File: routes/project.routes.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+function getProjects(request: Request, response: Response): void {
+    // eslint-disable-next-line no-console
+    console.log('hello world')
+}
+
+function getProject(request: Request, response: Response): void {
+    // eslint-disable-next-line no-console
+    console.log('bye bye')
```
Comment (Solenoden) [inline routes/project.routes.ts:15]: You forget a console log here

File: package.json [hunk 3]
```diff
@@ -33,7 +35,7 @@
     "@types/express": "4.17.1",
     "@types/morgan": "1.9.3",
     "eslint": "8.0.1",
-    "ts-node": "^10.4.0",
-    "typescript": "^4.5.2"
+    "ts-node": "10.4.0",
+    "typescript": "4.5.2"
```
Comment (Solenoden) [inline package.json:39]: I'm sure these can be devDependencies. The compiled javascript won't use these packages

**Review by Solenoden — CHANGES_REQUESTED** [review]

**Pull request merged**